Description: fixes incorrect output of the dump manipulator, when used on AVX2-enabled CPU (e.g. Intel Haswell)
Origin: http://www.boost.org/patches/1_55_0/001-log_fix_dump_avx2.patch
Reviewed-by: Steve Robbins <smr@debian.org>



Index: libs/log/src/dump_avx2.cpp
===================================================================
--- a/libs/log/src/dump_avx2.cpp	(revision 86614)
+++ b/libs/log/src/dump_avx2.cpp	(working copy)
@@ -205,7 +205,7 @@
             __m256i mm_input = _mm256_load_si256(reinterpret_cast< const __m256i* >(p));
             __m256i mm_output1, mm_output2, mm_output3;
             dump_pack(mm_char_10_to_a, mm_input, mm_output1, mm_output2, mm_output3);
-            store_characters_x3(mm_output1, mm_output2, mm_output3, buf);
+            store_characters_x3(mm_output1, mm_output2, mm_output3, b);
         }
 
         _mm256_zeroall(); // need to zero all ymm registers to avoid register spills/restores the compler generates around the function call
